### ISSUE TITLE
Support ghc 9.8

### ISSUE THREAD

**Issue by chenrui333** [lierdakil/pandoc-crossref#435]

**Comment by lierdakil** [comment]
Is there a reason to? Stackage is still on 9.6, and ghcup "recommended" is even older. It's not like it's policy or anything, but I kinda settled into lagging one major version behind GHC releases, because compiler bugs are quite annoying, and I've encountered my fair share of those with fresh major releases.

**Comment by lierdakil** [comment]
To clarify, I don't really have bandwidth to support multiple major GHC releases, so if I switch to 9.8, that's it, we're on 9.8 now.

**Comment by chenrui333** [comment]
yeah, I was mostly obligated to log an issue in here for visibility :) no strong reason, but I carry a bit different mindset like if the project is behind on the releases, it would be further behind kind of thing.

agree on the note of using older snd more stable releases whenever possible.

**Comment by lierdakil** [comment]
GHC 9.8 has hit stackage lts apparently, so I guess it's about time to make the jump. FWIW, the code did build (with warnings) on GHC 9.8, but I've resolved the warnings just now, too, so the current `master` should work.

I'll make a patch release in the near future. It should still be backwards-compatible with GHC 9.6, but I can make no guarantees going forward, so I will likely be tightening the bounds eventually.
